### PR TITLE
upgrade smithy to 1.27.2

### PR DESCRIPTION
**Pull request by syall**
*Issue #, if available:*

N/A

---

*Description of changes:*

Upgrade smithy to 1.27.2.

Tested with `make smithy-clean smithy-build smithy-publish-local`.

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: codegen/gradle.properties [hunk 1]
```diff
@@ -1,2 +1,2 @@
-smithyVersion=1.26.0
+smithyVersion=[1.27.0,1.28.0[
```
Comment (isaiahvita) [inline codegen/gradle.properties:1]: is this valid gradle syntax? with the hanging open bracket?

Comment (isaiahvita) [reply codegen/gradle.properties:1]: discussed offline, despite the latter way being more readable/explicit IMO, the former way written in the PR is more gradle-idiomatic. and i agree for most cases its better to follow convention. so im approving.

**Review by isaiahvita — DISMISSED** [review]

File: codegen/gradle.properties (hunk 1 above)
Comment (aajtodd) [reply codegen/gradle.properties:1]: Why a range in the first place? Why not a specific version here? We control the smithy version and when we upgrade, I'm not sure I'm understanding why we are declaring a range of versions to be compatible with.

Comment (syall) [reply codegen/gradle.properties]: Updated to `1.27.2`

**Review by lucix-aws — APPROVED** [review]

**Pull request merged**